apps/passport-client/src/dispatch.ts

```diff
@@ -942,3 +949,16 @@ async function promptToAgreePrivacyNotice(state: AppState, update: ZuUpdate) {
     });
   }
 }
+
+/**
+ * Merge in a PCD collection from an import of backed-up data.
+ */
+function mergeImport(
+  state: AppState,
+  update: ZuUpdate,
+  collection: PCDCollection,
+  filter: (pcd: PCD) => boolean
+) {
+  state.pcds.merge(collection, { filter, setFolders: true });
```

Comment:
FWIW, it worries me a bit that this isn't atomic, so if an exception were thrown the PCDs are in a partially-merged state.  Could be addressed by cloning the PCDCollection first, which I suspect I might implement for merging in background sync.

Reply:
Internally, `merge` calls `addAll`, which will not update the PCD collection if any of the `add` operations throws.

Reply:
I've added a cloning step though!